### PR TITLE
docs: improve the template tutorial's readability

### PR DESCRIPTION
**Pull request by lmgyuan**
**Description**

This PR is to make the tutorial documentation clearer for users to understand, as requested by @derberg : )

- Update the Overview of Steps to make it easier for users to understand what the project directory should look like.
- Add directory information so users know where to use what commands more clearly.
- Make the part that creates client.py clearer about where and how to use the commands.
- Add a detailed path for the directory of the TopicFunction.js file.
- Add three project directory graphs so users can have a visual understanding of the project structure. 

**Related issue(s)**
See also #1145

### CODE REVIEW

File: docs/generator-template.md [hunk 1]
```diff
@@ -67,10 +67,22 @@ components:
 
 1. Create a new directory for your template named **python-mqtt-client-template**.
 2. Install the AsyncAPI CLI using the command `npm install -g @asyncapi/cli`.
-3. Create a new folder **test/fixtures** with a file named **asyncapi.yml** in your fixtures directory. This file is used to define the **structure** of your template. 
-4. Create a new file named **package.json** in your template directory. This file is used to define the **dependencies** for your template.
-5. Create a new file named **index.js** in your **template** directory. This file is used to define the **logic** for your template.
-6. Create a **test.py** file to validate the logic of your application.
+3. Create a new folder **test/fixtures** with a file named **asyncapi.yml** in your fixtures directory. This file is used to define the **structure** of your template. You can copy the above template and paste it in your **asyncapi.yml** file.
```
Comment (derberg) [inline docs/generator-template.md:70]:
```suggestion
3. Create a new folder **test/fixtures** with a file named **asyncapi.yml** in your fixtures directory. This file is used to define the **structure** of your template. You can copy the above example and paste it in your **asyncapi.yml** document.
```

you mean the example AsyncAPI document, right? not the template?

File: docs/generator-template.md [hunk 2]
```diff
@@ -67,10 +67,22 @@ components:
...
+4. Create a new file named **package.json** in your **python-mqtt-client-template** directory. This file is used to define the **dependencies** for your template.
```
Comment (derberg) [inline docs/generator-template.md:71]: yup, good 👁️

File: docs/generator-template.md [hunk 3]
```diff
@@ -67,10 +67,22 @@ components:
...
+5. Create a new folder **python-mqtt-client-template/template**. Create a new file named **index.js** in your **template** directory. This file is used to define the **logic** for your template.
+6. Create a **test.py** file to validate the logic of your application. Don't worry about this file for now. The tutorial will tell you how to create it later.
+
+Now your directory should look like this:
+
+```
+python-mqtt-client-template
```
Comment (derberg) [inline docs/generator-template.md:78]: 👏🏼

File: docs/generator-template.md [hunk 4]
```diff
@@ -173,7 +185,7 @@ In this section, you'll:
 
 ### 1. Create the client
 
-The following is the sample code of the Python client you generated [above](#test-using-asyncapi-cli) using the Paho-MQTT library after running the `asyncapi generate fromTemplate test/fixtures/asyncapi.yml ./ -o test/project` command.
+Here is the sample code to be pasted in the client.py you generated above running the `asyncapi generate fromTemplate test/fixtures/asyncapi.yml ./ -o test/project` command. It uses the paho-mqtt package.
```
Comment (derberg) [inline docs/generator-template.md:188]:
```suggestion
Here is the sample code to be pasted in the client.py you generated above running the `asyncapi generate fromTemplate test/fixtures/asyncapi.yml ./ -o test/project` command. It uses the `paho-mqtt` package.
```

**Review by derberg — COMMENTED** [review]
thanks, tutorial got more explicit, I especially like the approach with showing directory structure

there are 2 comments @lmgyuan you need to address

**Review by Florence-Njeri — APPROVED** [review]

**Comment by derberg** [comment]
@allcontributors please add @lmgyuan for docs

**Pull request merged**